monai/bundle/scripts.py

```diff
@@ -51,62 +72,113 @@ def run(
-        target_id: ID name of the target component or workflow, it must have a `run` method.
         args_file: a JSON or YAML file to provide default values for `meta_file`, `config_file`,
-            `target_id` and override pairs. so that the command line inputs can be simplified.
+            `runner_id` and override pairs. so that the command line inputs can be simplified.
         override: id-value pairs to override or add the corresponding config content.
             e.g. ``--net#input_chns 42``.
 
     """
-    k_v = zip(["meta_file", "config_file", "target_id"], [meta_file, config_file, target_id])
-    for k, v in k_v:
-        if v is not None:
-            override[k] = v
-
-    full_kv = zip(
-        ("meta_file", "config_file", "target_id", "args_file", "override"),
-        (meta_file, config_file, target_id, args_file, override),
-    )
-    print("\n--- input summary of monai.bundle.scripts.run ---")
-    for name, val in full_kv:
-        print(f"> {name}: {pprint.pformat(val)}")
-    print("---\n\n")
 
-    _args = _update_default_args(args=args_file, **override)
+    _args = _update_args(args=args_file, runner_id=runner_id, meta_file=meta_file, config_file=config_file, **override)
     for k in ("meta_file", "config_file"):
         if k not in _args:
             raise ValueError(f"{k} is required for 'monai.bundle run'.\n{run.__doc__}")
+    _log_input_summary(tag="run", args=_args)
 
     parser = ConfigParser()
     parser.read_config(f=_args.pop("config_file"))
     parser.read_meta(f=_args.pop("meta_file"))
-    id = _args.pop("target_id", "")
+    id = _args.pop("runner_id", "")
 
-    # the rest key-values in the args are to override config content
+    # the rest key-values in the _args are to override config content
     for k, v in _args.items():
         parser[k] = v
 
     workflow = parser.get_parsed_content(id=id)
     if not hasattr(workflow, "run"):
         raise ValueError(f"The parsed workflow {type(workflow)} does not have a `run` method.\n{run.__doc__}")
     workflow.run()
+
+
+def verify_metadata(
+    meta_file: Optional[Union[str, Sequence[str]]] = None,
+    filepath: Optional[PathLike] = None,
+    result_path: Optional[PathLike] = None,
+    create_dir: Optional[bool] = None,
+    hash_val: Optional[str] = None,
+    args_file: Optional[str] = None,
+    **kwargs,
+):
+    """
+    Verify the provided `metadata` file based on the predefined `schema`.
+    `metadata` content must contain the `schema` field for the URL of shcema file to download.
+    The schema standard follows: http://json-schema.org/.
+
+    Args:
+        meta_file: filepath of the metadata file to verify, if `None`, must be provided in `args_file`.
+            if it is a list of file paths, the content of them will be merged.
+        filepath: file path to store the downloaded schema.
+        result_path: if not None, save the validation error into the result file.
```

Comment:
Why save the validation error here?

Reply:
Hi @wyli ,

If validation failed, the error message may be long, so I think it's useful to save in file.
Seems it's not necessary so far? Let me remove it.

Thanks.

Reply:
I see, that's fine, perhaps we could use logging to output to files though

Reply:
Sure, updated in the latest commit.
Also added regex filter to extract only the key information from long message:
```
--- input summary of monai.bundle.scripts.verify_metadata ---
2022-03-14 12:08:59,893 - INFO - > meta_file: '/tmp/tmplrx_tk2c/metadata.json'
2022-03-14 12:08:59,893 - INFO - > filepath: '/tmp/tmplrx_tk2c/schema.json'
2022-03-14 12:08:59,893 - INFO - ---


schema.json: 16.0kB [00:00, 23.1kB/s]
2022-03-14 12:09:00,604 - INFO - Downloaded: /tmp/tmplrx_tk2c/schema.json
2022-03-14 12:09:00,604 - INFO - Expected md5 is None, skip md5 check for file /tmp/tmplrx_tk2c/schema.json.
2022-03-14 12:09:00,744 - INFO - 'version' is a required property

Failed validating against schema `https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_202203130950.json`.
----------------------------------------------------------------------
```

Thanks.